### PR TITLE
fix(chat-e2e): fix compare mode regenerate test (Issue #924)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

Fix e2e test for compare mode regeneration

Issues:

- https://github.com/epam/ai-dial-chat/issues/924

PRs:

- https://github.com/epam/ai-dial-chat/pull/925

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-935.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-935.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/tests/compareMode.test.ts [hunk 1]
```diff
@@ -670,7 +670,7 @@ dialTest(
     await dialTest.step(
       'Click "Regenerate" button and verify last response is regenerated for both chats',
       async () => {
-        await chat.regenerate.waitForState();
+        await chat.regenerate.waitFirstForState();
```
Comment (irinakartun) [inline apps/chat-e2e/src/tests/compareMode.test.ts]: could you use:
`await chat.regenerate.getNthElement(1).waitFor()`
instead

File: apps/chat-e2e/src/ui/webElements/chat.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export class Chat extends BaseElement {
     const leftRequestPromise = this.waitForRequestSent(
       comparedEntities.leftEntity,
     );
-    await this.regenerate.click();
+    await this.regenerate.clickFirst();
```
Comment (irinakartun) [inline apps/chat-e2e/src/ui/webElements/chat.ts]: you could use:
`await this.regenerate.getNthElement(1).click();`

instead

File: apps/chat-e2e/src/ui/webElements/baseElement.ts [hunk 3]
```diff
@@ -77,6 +77,18 @@ export class BaseElement {
     await this.rootLocator.click(options);
   }
 
+  async clickFirst(options?: {
```
Comment (irinakartun) [inline apps/chat-e2e/src/ui/webElements/baseElement.ts]: please remove as there is easier solution

File: apps/chat-e2e/src/ui/webElements/baseElement.ts [hunk 4]
```diff
@@ -104,6 +116,13 @@ export class BaseElement {
     await this.rootLocator.waitFor(options);
   }
 
+  async waitFirstForState(options?: {
```
Comment (irinakartun) [inline apps/chat-e2e/src/ui/webElements/baseElement.ts]: please remove as there is easier solution

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-935.nightly-test.deltixhub.io
>E2E tests status: success

**Review by irinakartun — APPROVED** [review]

**Pull request merged**